packages/x-components/src/x-modules/queries-preview/store/__tests__/actions.spec.ts

```diff
@@ -0,0 +1,106 @@
+import Vuex, { Store } from 'vuex';
+import { createLocalVue } from '@vue/test-utils';
+import { SearchRequest } from '@empathyco/x-types';
+import { getSearchResponseStub } from '../../../../__stubs__/search-response-stubs.factory';
+import { getMockedAdapter, installNewXPlugin } from '../../../../__tests__/utils';
+import { SafeStore } from '../../../../store/__tests__/utils';
+import { queriesPreviewXStoreModule } from '../module';
+import {
+  QueriesPreviewState,
+  QueriesPreviewGetters,
+  QueriesPreviewMutations,
+  QueriesPreviewActions
+} from '../types';
+
+describe('testing queries preview module actions', () => {
+  const mockedSearchResponse = getSearchResponseStub();
+  const getQueryPreviewRequest = (query: string): SearchRequest => ({
+    query,
+    rows: 3,
+    extraParams: {
+      extraParam: 'extra param'
+    }
+  });
+
+  const adapter = getMockedAdapter({
+    search: mockedSearchResponse
+  });
+
+  const localVue = createLocalVue();
+  localVue.config.productionTip = false; // Silent production console messages.
+  localVue.use(Vuex);
+
+  const store: SafeStore<
+    QueriesPreviewState,
+    QueriesPreviewGetters,
+    QueriesPreviewMutations,
+    QueriesPreviewActions
+  > = new Store(queriesPreviewXStoreModule as any);
+  installNewXPlugin({ adapter, store }, localVue);
+
+  describe('fetchQueryPreview', () => {
+    it('should make a search request with a unique id', async () => {
+      const request = getQueryPreviewRequest('sandals');
```

Comment:
This test and the next one can be combined in one.

Reply:
Sure, I've combined them.